[kenzantonius] iP

kenzantonius opened the pull request:
## DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/nus-cs2103-AY2223S2/ip/pull/378).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Add Level 1

Commit:
Add Level 2

Commit:
Add Level 3

Commit:
Add Level 4

Commit:
Add level 5

Commit:
Add level 6

WilliamHaiweiGu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,143 @@
 public class Duke {
+    private Scanner scanner = new Scanner(System.in);
+    private ArrayList<Task> list = new ArrayList<>();
+    private void Input() {
```
Shouldn't this method name be in camelCase?

WilliamHaiweiGu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,143 @@
+    }
+
+    private String[] ErrorEventOrDeadline(String input, String textToReplace, String textToSplit) throws DukeException {
+        String[] splitInput = input.replaceFirst(textToReplace, "")
```
Maybe use plural form for array name?

WilliamHaiweiGu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,143 @@
+        }
+    }
+
+    private String[] ErrorEventOrDeadline(String input, String textToReplace, String textToSplit) throws DukeException {
```
Maybe use verbs and camelCase for method names? Say, "makeErrEventOrDdl"

WilliamHaiweiGu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,143 @@
+
 public class Duke {
+    private Scanner scanner = new Scanner(System.in);
+    private ArrayList<Task> list = new ArrayList<>();
```
Maybe use plural form in a collection name? Like "tasks"?

WilliamHaiweiGu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,143 @@
+                    continue;
+                }
+                if (userInput.startsWith("deadline")) {
+                    String[] deadline = ErrorEventOrDeadline(userInput, "deadline", "/by");
```
Maybe use plural form for array name?

WilliamHaiweiGu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,143 @@
+                    continue;
+                }
+                if (userInput.startsWith("event")) {
+                    String[] event = ErrorEventOrDeadline(userInput, "event", "/at");
```
Maybe use plural form for array name?

WilliamHaiweiGu reviewed (commented):
Coding standard is decent.
* Compliant with information hiding and encapsulation. No public class variables.
* Boolean variables start with "is".
* Indentation is strictly 4 whitespaces.
* Line length is never no longer than 120 chars.
* Strictly follows Egyptian style brackets and correct letter case for most names.
* Appropriate whitespaces within statements.
* All variables are initialized at declaration.

Some rooms for improvement:
* Minor naming errors.
* Should put every class in a package.
* _Documentation plz_ 🥺

GJ!

seadragon2000341 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    protected String by;
```
Maybe change name from by to something that is a noun

seadragon2000341 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,13 @@
+public class Event extends Task {
+    private String from;
```
similarly, conisder changing from to a noun

seadragon2000341 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
nice following convention

seadragon2000341 reviewed (commented):
Overall, good job! Code is neat and follows coding standards

Commit:
Add level 7

Commit:
Add level 8

Commit:
no message

Commit:
Merge branch 'branch-Level-7'

Commit:
no message

Commit:
Merge branch 'branch-Level-8'

Commit:
Make the code more OOP

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Divide classes into packages

Commit:
Merge branch 'branch-A-Packages'

Commit:
Merge branch 'add-gradle-support'

Commit:
Add JUnit tests to test the behavior of the code

Commit:
Add JavaDoc comments to the code.

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Tweak the code to comply with a coding standard.

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Implement Level 9. Give users a way to find a task by searching for a keyword.

Commit:
Merge branch 'branch-Level-9'

Commit:
Implement Level 10. Add a GUI to Duke.

Commit:
Merge branch 'branch-Level-10'

Commit:
Use Assertions.

Commit:
Improve code quality.

Command classes have some complicated expressions. This might confuse the reader of the code.

The expressions are changed into boolean variables with names that are self-explanatory so that the code can be easily understood.

Other parts of the code already follow the code quality, so no changes are made.

kenzantonius committed:
Merge pull request #2 from kenzantonius/branch-A-Assertions

Use Assertions.

kenzantonius committed:
Merge pull request #3 from kenzantonius/branch-A-CodeQuality

Improve code quality.

Commit:
Merge branch 'master' of https://github.com/kenzantonius/ip

Commit:
Add Ui.png.

kenzantonius committed:
Update README.md

Commit:
Merge branch 'master' of https://github.com/kenzantonius/ip

Commit:
Fix bugs.
